src/utils/postUrlParser.js

```diff
@@ -0,0 +1,75 @@
+export default url => {
+  const parseCatAuthorPermlink = u => {
+    const postRegex = /^https?:\/\/(.*)\/(.*)\/(@[\w.\d-]+)\/(.*)/i;
+    const postMatch = u.match(postRegex);
+    if (postMatch && postMatch.length === 5) {
+      return {
+        author: postMatch[3].replace('@', ''),
+        permlink: postMatch[4],
+      };
+    }
+    const authorRegex = /^https?:\/\/(.*)\/(.*)\/(@[\w.\d-]+)/i;
+    const authorMatch = u.match(authorRegex);
+    if (authorMatch && authorMatch.length === 4) {
+      return {
+        author: authorMatch[3].replace('@', ''),
+        permlink: null,
+      };
+    }
+    return null;
+  };
+
+  const parseAuthorPermlink = u => {
+    const r = /^https?:\/\/(.*)\/(@[\w.\d-]+)\/(.*)/i;
+    const match = u.match(r);
+    if (match && match.length === 4) {
+      return {
+        author: match[2].replace('@', ''),
+        permlink: match[3],
+      };
+    }
+    const authorRegex = /^https?:\/\/(.*)\/(@[\w.\d-]+)/i;
+    const authorMatch = u.match(authorRegex);
+    if (authorMatch && authorMatch.length === 3) {
+      return {
+        author: authorMatch[2].replace('@', ''),
+        permlink: null,
+      };
+    }
+
+    return null;
+  };
+
+  if (
+    ['https://esteem.app', 'https://steemit.com', 'https://steempeak.com'].some(x =>
+      url.startsWith(x),
+    )
+  ) {
```

Comment:
What’s difference here, parseCatAuthorpermlink and parseAuthorpermlink?

Reply:
parseCatAuthorpermlink for baseUrl + category + author + permlink
parseAuthorpermlink for baseUrl + author + permlink.

We are using same code with surfer side

Reply:
oh I see 👍